### DISCUSSION TITLE
posts/tarea-xsl/

### DISCUSSION THREAD

**Discussion by giscus** [marcosruiz/marcosruiz.github.io#15] [General]
# posts/tarea-xsl/

Entrega y presentación

https://marcosruiz.github.io/posts/tarea-xsl/

**Comment by rodrigomulero** [comment]
Recomiendo usar la hta multimarkdown para la transformación de Markdown a html, nos genera un código con bloques de código escapados y sin css, no nos incluye la cabecera del html pero es lo de menos. Pandoc no lo recomiendo porque genera los html con estilos css lo cual es incompatible con xsl-fo.

La hta usada en mi caso para pasar html a fo es saxon, a través de gestor de paquetes en macOS con homebrew.

La hta usada para pasar el fo a pdf es Apache FOP, instalada también a través de gestor de paquetes homebrew.

La incorporación de metadatos como título o autor aunque no es compatible con todos los lectores de pdf (son con adobe lógicamente) se realiza con Apache FOP incorporando XMP (eXtensible Metadata Platform) https://xmlgraphics.apache.org/fop/1.1/metadata.html

Para que los bloques de código mantengan su formato:

```xml
    <xsl:template match="code">
        <fo:block
            white-space-collapse="false" linefeed-treatment="preserve"
            white-space-treatment="preserve">
            <xsl:apply-templates />
        </fo:block>
    </xsl:template>
```

Pero recordad que es muy importante escapar:

```xml
<code class="xml">&lt;ciclo id=&quot;SMR&quot;&gt;
      &lt;nombre&gt;Sistemas Microinformáticos y Redes&lt;/nombre&gt;
      &lt;grado&gt;Medio&lt;/grado&gt;
      &lt;decretoTitulo año=&quot;2008&quot;/&gt;
    &lt;/ciclo&gt;
</code>
``

**Comment by lensseoyhshi** [comment]
¡Mis contribuciones en la[1](https://www.ai2markdown.com/markdown-to-html) son realmente valiosas! No solo proporciono una variedad de ejercicios de XSLT, sino que también muestro de manera detallada cómo transformar documentos XML en HTML y finalmente en PDF. Estos recursos son ideales tanto para principiantes como para aquellos que buscan profundizar en el tema. La estructura clara y los ejemplos detallados facilitan el aprendizaje autónomo y el desarrollo de habilidades en XSLT. ¡Estoy muy orgulloso de poder apoyar a quienes desean aprender y mejorar en esta área!
